service/src/main/java/bio/terra/workspace/service/job/JobBuilder.java

```diff
@@ -116,29 +126,80 @@ public JobBuilder addParameter(String keyName, @Nullable Object val) {
   /**
    * Submit a job to stairway and return the jobID immediately.
    *
+   * @param changeType Specify the type of {@link ActivityLogChangedType} to be logged.
+   * @param resultClass type for the response of the flight. When null, nothing is returned from the
+   *     flight.
    * @return jobID of submitted flight
    */
-  public String submit() {
+  public <T> String submit(
+      @Nullable ActivityLogChangedType changeType, @Nullable Class<T> resultClass) {
     populateInputParams();
-    return jobService.submit(flightClass, jobParameterMap, jobId);
+    if (changeType == null) {
+      return jobService.submit(flightClass, jobParameterMap, jobId);
+    }
+    return jobService.submitWithCallback(
+        flightClass,
+        resultClass,
+        jobParameterMap,
+        jobId,
+        new FutureCallback<>() {
+          @Override
+          public void onSuccess(JobResultOrException<T> result) {
+            if (changeType.equals(ActivityLogChangedType.DELETE)) {
+              activityLogDao.setChangedDate(workspaceId, changeType);
+            } else {
+              if (result.getException() == null) {
```

Comment:
I know this code is moving, these comments apply to new place

// Only log to activity log if flight succeeded.